app/coffee/DocArchiveManager.coffee

```diff
@@ -0,0 +1,67 @@
+MongoManager = require "./MongoManager"
+MongoAWS = require "./MongoAWS"
+LockManager = require "./LockManager"
+logger = require "logger-sharelatex"
+_ = require "underscore"
+async = require "async"
+settings = require("settings-sharelatex")
+request = require("request")
+crypto = require("crypto")
+thirtySeconds = 30 * 1000
+
+module.exports = DocArchiveManager =
+
+	archiveAllDocsChanges: (project_id, callback = (error, docs) ->) ->
+		if settings.filestore?.backend != "s3"
+			return callback(null)
+		MongoManager.getProjectsDocs project_id, (error, docs) ->
```

Comment:
Could we please change this so it asks the docstore rather than mongo for the projects docs? This way is fine but its nice at the moment how only the docstore touches the doc collection. The request is just:

get "#{settings.internal.apis.docstore}/project/#{project_id}/doc" which will return you basically the same thing in json.

Maybe put it in a DocstoreHandler.coffee or something similar?

Same goes for unArchiveAllDocsChanges below.